# Generated by Django 4.0.6 on 2022-07-07 05:32
from django.contrib.auth.hashers import make_password
from django.db import migrations


def _add_default_users(apps, schema_editor):
    User = apps.get_model('users', 'User')
    for username, raw_password in [
        ('test_user@meistery.net', 'trial_application'),
        ('test_user2@meistery.net', 'trial_application')
    ]:
        User.objects.get_or_create(
            email=username,
            defaults={'password': make_password(raw_password), 'is_staff': False, 'is_superuser': False}
        )


def _add_default_application(apps, schema_editor):
    Application = apps.get_model('oauth2_provider', 'Application')
    Application.objects.get_or_create(
        name='Trial Application',
        defaults={
            'client_id': 'C8Qp0eQnhosJ9Z1DNd7UvQ5M9tiMT4pleqlWkqW3',
            'client_secret': 'b49eoIqNsMEtUhlGjymYabD14AbfCWx99Z0bcsO82AYweB9ZaZJJtZARmDx1f1QN3kjy7IpKYuuZzynUy7NDpeRVZcFPA8XaHR1pAXvVeBNMh9dxvZjqknSrqvwzDkRd',
            'client_type': 'public',
            'authorization_grant_type': 'password',
        }
    )


def add_default_credentials(apps, schema_editor):
    """
    Creating users with this usages(test) is not the right way.
    This usually happens in the devops process.
    """
    _add_default_users(apps, schema_editor)
    _add_default_application(apps, schema_editor)


class Migration(migrations.Migration):
    dependencies = [
        ('users', '0001_initial'),
        ('oauth2_provider', '0006_alter_application_client_secret'),
    ]

    operations = [
        migrations.RunPython(add_default_credentials, reverse_code=migrations.RunPython.noop),
    ]
